compiler-plugin/src/main/java/io/ballerina/wso2/apim/catalog/codemodifier/OpenAPIAnnotationModifier.java

```diff
@@ -0,0 +1,249 @@
+        for (AnnotationNode annotation: currentAnnotations) {
+            if (isServiceCatalogConfigAnnotationAvailable(annotation)) {
+                serviceCatalogConfigAnnotationUpdated = true;
+                SeparatedNodeList<MappingFieldNode> updatedFields = getUpdatedFields(annotation, openApiDefinition);
+                MappingConstructorExpressionNode annotationValue =
+                        annotation.annotValue().get().modify().withFields(updatedFields).apply();
+                annotation = annotation.modify().withAnnotValue(annotationValue).apply();
+            }
+            updatedAnnotations = updatedAnnotations.add(annotation);
+        }
+        if (!serviceCatalogConfigAnnotationUpdated) {
+            AnnotationNode openApiAnnotation = getServiceCatalogConfigAnnotation(openApiDefinition);
+            updatedAnnotations = updatedAnnotations.add(openApiAnnotation);
+        }
+        return updatedAnnotations;
+    }
+
+    private SeparatedNodeList<MappingFieldNode> getUpdatedFields(AnnotationNode annotation, String openAPIDef) {
+        Optional<MappingConstructorExpressionNode> annotationValueOpt = annotation.annotValue();
+        if (annotationValueOpt.isEmpty()) {
+            return NodeFactory.createSeparatedNodeList(createOpenApiDefinitionField(openAPIDef));
+        }
+        List<Node> fields = new ArrayList<>();
+        MappingConstructorExpressionNode annotationValue = annotationValueOpt.get();
+        SeparatedNodeList<MappingFieldNode> existingFields = annotationValue.fields();
+        Token separator = NodeFactory.createToken(SyntaxKind.COMMA_TOKEN);
+        boolean openApiDefAvailable = false;
+        int existingFieldLength = existingFields.size();
+        int fieldIndex = 0;
+        for (MappingFieldNode field : existingFields) {
+            if (field instanceof SpecificFieldNode) {
+                String fieldName = ((SpecificFieldNode) field).fieldName().toString();
+                if (Constants.OPEN_API_DEFINITION_FIELD.equals(fieldName.trim())) {
+                    openApiDefAvailable = true;
+                    field = createOpenApiDefinitionField(openAPIDef);
+                    fields.add(field);
+                    fields.add(separator);
+                    continue;
+                }
+            }
+            fields.add(field);
+            if (existingFieldLength > 1 && fieldIndex < existingFieldLength - 1) {
+                fields.add(existingFields.getSeparator(fieldIndex++));
+            }
+        }
+        if (openApiDefAvailable) {
+            if (fields.size() != 0) {
+                fields.remove(fields.size() - 1);
+            }
+        } else {
+            fields.add(createOpenApiDefinitionField(openAPIDef));
+        }
+        return NodeFactory.createSeparatedNodeList(fields);
+    }
+
+    private String generateOASForgetServiceDeclarationNode(ServiceDeclarationNode serviceDeclarationNode,
```

Comment:
Bit confused about the function name, whether it is `Forget` or `ForGet`. If this `generateOASForgetServiceDeclarationNode` function is for generating an OAS spec, Shall we change the function name to reflect its behavior? 
ex: `generateOASForGivenServiceDeclarationNode`, `generateOASForServiceNode`.